wgsl/wgsl_unit_tests_template.py

```diff
@@ -85,3 +87,58 @@
     Case ("alias z = a<1<<c>;",name="template arg after 1 << is shift"),
     Case ("alias z = a<1<<c<d>()>;",name="template arg after 1 << is shift followed by templated value constructor"),
 ]
+
+UNMATCHED = ''
+match_cases = [
+    MatchCase("const z = a<b;","template_list",UNMATCHED,name="exposed <"),
+    MatchCase("const z = a>b;","template_list",UNMATCHED,name="exposed >"),
+    MatchCase("const z = (a<b)>c;","template_list",UNMATCHED,name="nested initial <"),
+    MatchCase("const z = a<(b>c);","template_list",UNMATCHED,name="nested final >"),
+
+    MatchCase("const z = a( b<c,  d>(e));","template_list","template_list:<c,  d>"),
+    MatchCase("const z = a((b<c), d>(e));","template_list",UNMATCHED,"nested initial <, exposed final >"),
+
+    MatchCase("const z = a<b[c];","temlate_list",UNMATCHED),
+    MatchCase("const z = a<b[select(1,2,c>(d))];","template_list",UNMATCHED),
+    MatchCase("const z = a<b[c>(d)];","template_list",UNMATCHED),
+    MatchCase("fn z() {if a < b {} else if c > d {}}","template_list",UNMATCHED,name="stop at braces"),
+
+    # Test interaction with lower-precedence expression operators: && and ||
+    MatchCase("const z = a<b&&c>(d);","template_list",UNMATCHED,name="&&"),
+    MatchCase("const z = a<b||c>(d);","template_list",UNMATCHED,name="||"),
+    MatchCase("const z = a<(b&&c)>(d);","template_list","template_list:<(b&&c)>",name="(&&)"),
+    MatchCase("const z = a<(b||c)>(d);","template_list","template_list:<(b||c)>",name="(||)"),
+
+    MatchCase("const z = a<b>();","template_list","template_list:<b>",name="templated value constructor"),
+    MatchCase("alias z = array<vec3<i32>,5>;","template_list","template_list:<vec3<i32>,5>",name="nested outer"),
+    MatchCase("alias z = array<vec3<i32>,5>;","//template_list//template_list","template_list:<i32>",name="nested inner"),
+    MatchCase("const z = a<1+2>();","template_list","template_list:<1+2>"),
+    MatchCase("const z = a<1,b>();","template_list","template_list:<1,b>"),
+    MatchCase("const z = a<b,c>==d;","template_list","template_list:<b,c>"),
+
+    # Check valid parses with things that start with '<' in a template list.
+
+    # This is a comparison of a less-than b<c>()
+    MatchCase("const z = a<b<c>();","ident","ident:z ident:a ident:b ident:c"),
+    MatchCase("const z = a<b<c>();","template_elaborated_ident","template_elaborated_ident:a template_elaborated_ident:b<c>"),
+    MatchCase("const z = a<b<c>();","template_list","template_list:<c>"),
+    # This is a nested template list
+    MatchCase("const z = a<b<c>>();","template_list","template_list:<b<c>>"),
+    MatchCase("const z = a<b<c>>();","template_list//template_list","template_list:<c>"),
+    MatchCase("const z = a<b<c>()>();","template_list","template_list:<b<c>()>"),
+    MatchCase("const z = a<b<c>()>();","template_list//template_list","template_list:<c>"),
+    # Check '<='
+    MatchCase("alias z = a<b<=c>;",  "template_list","template_list:<b<=c>",name="template arg <="),
```

Comment:
e.g. this says there is a template_list in the parse tree, corresponding to `<b<=c>` in the source text.  
The point is that the code points `<=` is recognized as an operator in the middle of an expression `b<=c`.